code/datums/armor.dm

```diff
@@ -0,0 +1,67 @@
+#define ARMORID "armor-[melee]-[bullet]-[laser]-[energy]-[bomb]-[bio]-[rad]-[fire]-[acid]"
+
+/proc/getArmor(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 0, acid = 0)
+  . = locate(ARMORID)
+  if (!.)
+    . = new /datum/armor(melee, bullet, laser, energy, bomb, bio, rad, fire, acid)
+
+/datum/armor
+  var/melee
+  var/bullet
+  var/laser
+  var/energy
+  var/bomb
+  var/bio
+  var/rad
+  var/fire
+  var/acid
+  datum_flags = DF_USE_TAG
```

Comment:
Move this up to the top please, I try to enforce new vars coming last followed by statics